src/meta/src/manager/catalog/mod.rs

```diff
@@ -838,64 +840,305 @@ where
+        for table in database_mgr.tables.values() {
+            if table.dependent_relations.contains(&relation_id) {
+                let mut table = table.clone();
+                table.definition = alter_relation_rename_refs(&table.definition, from, to);
+                to_update_tables.push(table);
+            }
+        }
+
+        for view in database_mgr.views.values() {
+            if view.dependent_relations.contains(&relation_id) {
+                let mut view = view.clone();
+                view.sql = alter_relation_rename_refs(&view.sql, from, to);
+                to_update_views.push(view);
+            }
+        }
+
+        for sink in database_mgr.sinks.values() {
+            if sink.dependent_relations.contains(&relation_id) {
+                let mut sink = sink.clone();
+                sink.definition = alter_relation_rename_refs(&sink.definition, from, to);
+                to_update_sinks.push(sink);
+            }
         }
+
+        // commit meta.
+        let mut tables = BTreeMapTransaction::new(&mut database_mgr.tables);
+        let mut views = BTreeMapTransaction::new(&mut database_mgr.views);
+        let mut sinks = BTreeMapTransaction::new(&mut database_mgr.sinks);
+        let mut sources = BTreeMapTransaction::new(&mut database_mgr.sources);
+        to_update_tables.iter().for_each(|table| {
+            tables.insert(table.id, table.clone());
+        });
+        // TODO: there are some inconsistencies in the process of notifying the frontend, we need to
+        // support batch notification.
+        to_update_views.iter().for_each(|view| {
+            views.insert(view.id, view.clone());
+        });
+        to_update_sinks.iter().for_each(|sink| {
+            sinks.insert(sink.id, sink.clone());
+        });
+        if let Some(source) = &to_update_source {
+            sources.insert(source.id, source.clone());
+        }
+        commit_meta!(self, tables, views, sinks, sources)?;
+
+        // 5. notify frontend.
+        let mut version = 0;
+        assert!(
+            !to_update_tables.is_empty()
+                || !to_update_views.is_empty()
+                || !to_update_sinks.is_empty()
+                || to_update_source.is_some()
+        );
+        for table in to_update_tables {
+            version = self
+                .notify_frontend(Operation::Update, Info::Table(table))
+                .await;
+        }
+        for view in to_update_views {
+            version = self
+                .notify_frontend(Operation::Update, Info::View(view))
+                .await;
+        }
+        for sink in to_update_sinks {
+            version = self
+                .notify_frontend(Operation::Update, Info::Sink(sink))
+                .await;
+        }
+        if let Some(source) = to_update_source {
+            version = self
+                .notify_frontend(Operation::Update, Info::Source(source))
+                .await;
+        }
```

Comment:
We can use group notification to replace this snippet after https://github.com/risingwavelabs/risingwave/pull/8741 gets merged.